specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json

```diff
@@ -0,0 +1,340 @@
+        },
+        "orderby": {
+          "x-ms-client-name": "OrderBy",
+          "type": "string",
+          "description": "The comma-separated list of OData $orderby expressions by which to sort the results. Each expression can be either a field name or a call to the geo.distance() function. Each expression can be followed by asc to indicate ascending, and desc to indicate descending. The default is ascending order. Ties will be broken by the match scores of documents. If no OrderBy is specified, the default sort order is descending by document match score. There can be at most 32 Orderby clauses."
+        },
+        "search": {
+          "type": "string",
+          "description": "The search text on which to base suggestions."
+        },
+        "searchFields": {
+          "type": "string",
+          "description": "The comma-separated list of field names to consider when querying for suggestions."
+        },
+        "select": {
+          "type": "string",
+          "description": "The comma-separated list of fields to retrieve. If unspecified, all fields marked as retrievable in the schema are included."
+        },
+        "suggesterName": {
+          "type": "string",
+          "description": "The name of the suggester as specified in the suggesters collection that's part of the index definition."
+        },
+        "top": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of suggestions to retrieve. This must be a value between 1 and 100. The default is to 5."
+        }
+      },
+      "description": "Parameters for filtering, sorting, fuzzy matching, and other suggestions query behaviors."
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Client Api Version."
+    },
+    "ClientRequestIdParameter": {
+      "name": "client-request-id",
+      "in": "header",
+      "required": false,
+      "type": "string",
+      "format": "uuid",
+      "description": "The tracking ID sent with the request to help with debugging.",
+      "x-ms-client-request-id": true,
+      "x-ms-parameter-grouping": { "name": "search-request-options" },
+      "x-ms-parameter-location": "method"
+    },
+    "SearchServiceNameParameter": {
+      "name": "searchServiceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "description": "The name of the Azure Search service.",
+      "x-ms-parameter-location": "client"
+    },
+    "SearchDnsSuffixParameter": {
+      "name": "searchDnsSuffix",
```

Comment:
This required parameter is not provided in any example.

Reply:
Do downstream consumers of these examples do anything useful with x-ms-parameterized-host parameters like this one? For example, when we start generating REST API documentation from Swagger (we aren't yet for our data plane, for historical reasons), will the examples in the docs put this parameter and `searchServiceName` in the host name part of the request URL where it should be?

Reply:
The doc team is using the x-ms-examples for generating examples in the docs. I think they are doing that. [Here](https://docs.microsoft.com/en-us/rest/api/storagerp/srp_json_create_storage_account) is an example for the Storage Management API. Please contact the doc team for feature requests or any feedback.

Reply:
@amarzavery Do you happen to know if [these KeyVault docs](https://docs.microsoft.com/rest/api/keyvault/createkey/createkey) are auto-generated as well? This seems to be an example of what we're hoping for.